src/fava/core/query.py

```diff
@@ -0,0 +1,167 @@
+"""Query result types."""
+
+from __future__ import annotations
+
+import datetime
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import TYPE_CHECKING
+
+from beancount.core.amount import Amount
+from beancount.core.inventory import Inventory
+from beancount.core.position import Position
+
+from fava.core.conversion import simple_units
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Literal
+    from typing import TypeAlias
+    from typing import TypeVar
+
+    from fava.core.inventory import SimpleCounterInventory
+
+    T = TypeVar("T")
+
+    QueryRowValue = (
+        bool | int | str | datetime.date | Decimal | Position | Inventory
+    )
+    SerialisedQueryRowValue = (
+        bool
+        | int
+        | str
+        | datetime.date
+        | Decimal
+        | Position
+        | SimpleCounterInventory
+    )
```

Comment:
There are a few types that can be returned by a `beanquery` query missing here.

Reply:
There are some missing ones indeed. It's just the column types that Fava currently send in some serialisation to the frontend - everything else just get's serialised to a string.